packages/admin/admin/src/dataGrid/useDataGridRemote.test.ts

```diff
@@ -1,8 +1,202 @@
+        });
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        expect(result.current.pageSize).toBe(mockedPageSize);
+    });
+
+    it("writes sort param to URL", () => {
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onSortModelChange?.([{ field: mockedSortField, sort: mockedSort }], {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ sort: `${mockedSortField}:${mockedSort}` }) });
+    });
+
+    it("writes filter param to URL", () => {
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onFilterModelChange?.(mockedFilterModel, {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ filter: JSON.stringify(mockedFilterModel) }) });
+    });
+
+    it("writes page param to URL", () => {
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onPageChange?.(mockedPage, {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ page: mockedPage }) });
+    });
+
+    it("writes pageSize param to URL", () => {
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote());
+
+        result.current.onPageSizeChange?.(mockedPageSize, {});
+
+        expect(mockedReplace).toHaveBeenCalledWith({ search: queryString.stringify({ pageSize: mockedPageSize }) });
+    });
+
+    it("uses queryParamsPrefix when reading params from URL", () => {
+        const prefix = "mocked";
+        (useLocation as jest.Mock).mockReturnValue({
+            search: queryString.stringify({
+                [`${prefix}sort`]: `${mockedSortField}:${mockedSort}`,
+                [`${prefix}filter`]: JSON.stringify(mockedFilterModel),
+                [`${prefix}page`]: mockedPage,
+                [`${prefix}pageSize`]: mockedPageSize,
+            }),
+        });
+
+        const { result } = renderHook(() => useDataGridRemote({ queryParamsPrefix: prefix }));
+
+        expect(result.current.sortModel).toEqual([{ field: mockedSortField, sort: mockedSort }]);
+        expect(result.current.filterModel).toEqual(mockedFilterModel);
+        expect(result.current.page).toBe(mockedPage);
+        expect(result.current.pageSize).toBe(mockedPageSize);
+    });
+
+    it("adds queryParamsPrefix when writing to URL if queryParamsPrefix is set", () => {
+        const prefix = "mocked";
+        const mockedReplace = mockReplace();
+
+        const { result } = renderHook(() => useDataGridRemote({ queryParamsPrefix: prefix }));
+
+        result.current.onSortModelChange?.([{ field: mockedSortField, sort: mockedSort }], {});
+        expect(mockedReplace.mock.calls[0]?.[0]?.search).toContain(`${prefix}sort`);
```

Comment:
Why can't you use `toHaveBeenCalledWith` for this test?

Reply:
With `toHaveBeenCalledWith`, I need to know the exact structure of the passed argument (at least to my knowledge). That's why I opted for this approach because here I can use `toContain`